webapp/app/head_to_head/controllers/headToHead.controller.js

```diff
@@ -7,7 +7,18 @@ angular.module('app.headToHead').controller("HeadToHeadController", function($sc
     $scope.wins = 0;
     $scope.losses = 0;
 
-    $scope.onChange = function() {
+    // get fresh "player" from the change listener, since the model updates after the change listener is fired...
```

Comment:
this was an unfortunate thing that I didn't know how to get around without these callbacks